ci-operator/step-registry/azure/provision/resourcegroup/cross-subscription/azure-provision-resourcegroup-cross-subscription-commands.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# save the exit code for junit xml file generated in step gather-must-gather
+# pre configuration steps before running installation, exit code 100 if failed,
+# save to install-pre-config-status.txt
+# post check steps after cluster installation, exit code 101 if failed,
+# save to install-post-check-status.txt
+EXIT_CODE=100
+trap 'if [[ "$?" == 0 ]]; then EXIT_CODE=0; fi; echo "${EXIT_CODE}" > "${SHARED_DIR}/install-pre-config-status.txt"' EXIT TERM
+
+RG_NAME="${NAMESPACE}-${UNIQUE_HASH}-rg-cross-sub"
+REGION="${HYPERSHIFT_AZURE_LOCATION:-${LEASED_RESOURCE}}"
+echo "Azure region: ${REGION}"
+
+# az should already be there
+command -v az
+az --version
+
+# set the parameters we'll need as env vars
+if [[ -f "${SHARED_DIR}/azure-sp-contributor.json" ]]; then
+  echo "Setting AZURE credential with Contributor role only for installer"
+  AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/azure-sp-contributor.json" 
+else
+  echo "Setting AZURE credential with  default SP for installer"
+  AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
```

Comment:
Are you sure this step with default SP can be performed successfully? Maybe it's safe to exit here (exit 0) to avoid noise on old ocp version when the step is added into existing workflow.

Reply:
Good point!
Currently the default SP doesn't have the permission in the cross subscription, so I think we should "exit 1", right?

Reply:
If you don't plan to add this step into existing workflow to cover more complex configurtion, then I'm okay with "exit 1".

Reply:
okay